NegativeArraySizeException when weaving aspect containing non-public, overloaded methods

I have a Spring Boot app that uses AspectJ. I am building with Maven. My `mvn package` command fails with
```text
AspectJ Internal Error: unable to add stackmap attributes to class 'gov.nasa.drf.auth.accesscontrol.AuthorizationAspect'. -1
java.lang.NegativeArraySizeException: -1
      at aj.org.objectweb.asm.Frame.merge(Frame.java:1222)
      at aj.org.objectweb.asm.MethodWriter.computeAllFrames(MethodWriter.java:1611)
      at aj.org.objectweb.asm.MethodWriter.visitMaxs(MethodWriter.java:1547)
      at aj.org.objectweb.asm.MethodVisitor.visitMaxs(MethodVisitor.java:786)
      at aj.org.objectweb.asm.ClassReader.readCode(ClassReader.java:2665)
      at aj.org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1514)
      at aj.org.objectweb.asm.ClassReader.accept(ClassReader.java:744)
      at aj.org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
      at org.aspectj.weaver.bcel.asm.StackMapAdder.addStackMaps(StackMapAdder.java:45)
      at org.aspectj.weaver.bcel.LazyClassGen.getJavaClassBytesIncludingReweavable(LazyClassGen.java:762)
      at org.aspectj.weaver.bcel.BcelWeaver.getClassFilesFor(BcelWeaver.java:1459)
      at org.aspectj.weaver.bcel.BcelWeaver.weaveAndNotify(BcelWeaver.java:1421)
      at org.aspectj.weaver.bcel.BcelWeaver.weave(BcelWeaver.java:1161)
      at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.weaveQueuedEntries(AjPipeliningCompilerAdapter.java:510)
      at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.afterCompiling(AjPipeliningCompilerAdapter.java:374)
      at org.aspectj.ajdt.internal.compiler.CompilerAdapter.ajc$afterReturning$org_aspectj_ajdt_internal_compiler_CompilerAdapter$2$f9cc9ca0(CompilerAdapter.aj:79)
      at org.aspectj.org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:428)
      at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performCompilation(AjBuildManager.java:1101)
      at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performBuild(AjBuildManager.java:275)
      at org.aspectj.ajdt.internal.core.builder.AjBuildManager.batchBuild(AjBuildManager.java:188)
      at org.aspectj.ajdt.ajc.AjdtCommand.doCommand(AjdtCommand.java:103)
      at org.aspectj.ajdt.ajc.AjdtCommand.runCommand(AjdtCommand.java:47)
      at org.aspectj.tools.ajc.Main.run(Main.java:374)
      at org.aspectj.tools.ajc.Main.runMain(Main.java:253)
      at org.codehaus.mojo.aspectj.AbstractAjcCompiler.execute(AbstractAjcCompiler.java:568)
      at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
      at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
      at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
      at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
      at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
      at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
      at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
      at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
      at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
      at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
      at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
      at java.base/java.lang.reflect.Method.invoke(Method.java:568)
      at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
      at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
      at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
      at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[INFO] Join point 'method-execution(org.springframework.http.ResponseEntity gov.nasa.drf.auth.controller.HealthController.getDatabaseHealth(java.lang.String))' in Type 'gov.nasa.drf.auth.controller.HealthController' (HealthController.java:41) advised by around advice from 'gov.nasa.drf.auth.accesscontrol.AuthenticationAspect' (AuthenticationAspect.class(from AuthenticationAspect.java))
[ERROR] ABORT
Jul 31, 2023 8:13:42 PM org.aspectj.weaver.tools.Jdk14Trace info
INFO: Dumping to /buildDir/./ajcore.20230731.201342.265.txt
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:17 min
[INFO] Finished at: 2023-07-31T20:13:42Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:aspectj-maven-plugin:1.14.0:compile (default-compile) on project auth: AJC compiler errors:
[ERROR] abort ABORT -- (BCException) Unable to find ASM classes (org.objectweb.asm.ClassReader, org.objectweb.asm.ClassVisitor) for stackmap generation. Stackmap generation for woven code is required to avoid verify errors on a Java 1.7 or higher runtime.
[ERROR] when weaving classes
[ERROR] when weaving
[ERROR] when batch building BuildConfig[null] #Files=0 AopXmls=#0
[ERROR]
[ERROR] Unable to find ASM classes (org.objectweb.asm.ClassReader, org.objectweb.asm.ClassVisitor) for stackmap generation. Stackmap generation for woven code is required to avoid verify errors on a Java 1.7 or higher runtime.
[ERROR] when weaving classes
[ERROR] when weaving
[ERROR] when batch building BuildConfig[null] #Files=0 AopXmls=#0
[ERROR]
[ERROR] org.aspectj.weaver.BCException: Unable to find ASM classes (org.objectweb.asm.ClassReader, org.objectweb.asm.ClassVisitor) for stackmap generation. Stackmap generation for woven code is required to avoid verify errors on a Java 1.7 or higher runtime.
[ERROR] when weaving classes
[ERROR] when weaving
[ERROR] when batch building BuildConfig[null] #Files=0 AopXmls=#0
[ERROR]
[ERROR]       at org.aspectj.weaver.bcel.LazyClassGen.getJavaClassBytesIncludingReweavable(LazyClassGen.java:756)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.getClassFilesFor(BcelWeaver.java:1459)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.weaveAndNotify(BcelWeaver.java:1421)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.weave(BcelWeaver.java:1192)
[ERROR]       at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.weaveQueuedEntries(AjPipeliningCompilerAdapter.java:510)
[ERROR]       at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.afterCompiling(AjPipeliningCompilerAdapter.java:374)
[ERROR]       at org.aspectj.ajdt.internal.compiler.CompilerAdapter.ajc$afterReturning$org_aspectj_ajdt_internal_compiler_CompilerAdapter$2$f9cc9ca0(CompilerAdapter.aj:79)
[ERROR]       at org.aspectj.org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:428)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performCompilation(AjBuildManager.java:1101)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performBuild(AjBuildManager.java:275)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.batchBuild(AjBuildManager.java:188)
[ERROR]       at org.aspectj.ajdt.ajc.AjdtCommand.doCommand(AjdtCommand.java:103)
[ERROR]       at org.aspectj.ajdt.ajc.AjdtCommand.runCommand(AjdtCommand.java:47)
[ERROR]       at org.aspectj.tools.ajc.Main.run(Main.java:374)
[ERROR]       at org.aspectj.tools.ajc.Main.runMain(Main.java:253)
[ERROR]       at org.codehaus.mojo.aspectj.AbstractAjcCompiler.execute(AbstractAjcCompiler.java:568)
[ERROR]       at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]       at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
[ERROR]       at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
[ERROR]       at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]       at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]       at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
[ERROR]       at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
[ERROR]       at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
[ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] Caused by: java.lang.NegativeArraySizeException: -1
[ERROR] java.lang.NegativeArraySizeException: -1
[ERROR]       at aj.org.objectweb.asm.Frame.merge(Frame.java:1222)
[ERROR]       at aj.org.objectweb.asm.MethodWriter.computeAllFrames(MethodWriter.java:1611)
[ERROR]       at aj.org.objectweb.asm.MethodWriter.visitMaxs(MethodWriter.java:1547)
[ERROR]       at aj.org.objectweb.asm.MethodVisitor.visitMaxs(MethodVisitor.java:786)
[ERROR]       at aj.org.objectweb.asm.ClassReader.readCode(ClassReader.java:2665)
[ERROR]       at aj.org.objectweb.asm.ClassReader.readMethod(ClassReader.java:1514)
[ERROR]       at aj.org.objectweb.asm.ClassReader.accept(ClassReader.java:744)
[ERROR]       at aj.org.objectweb.asm.ClassReader.accept(ClassReader.java:424)
[ERROR]       at org.aspectj.weaver.bcel.asm.StackMapAdder.addStackMaps(StackMapAdder.java:45)
[ERROR]       at org.aspectj.weaver.bcel.LazyClassGen.getJavaClassBytesIncludingReweavable(LazyClassGen.java:762)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.getClassFilesFor(BcelWeaver.java:1459)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.weaveAndNotify(BcelWeaver.java:1421)
[ERROR]       at org.aspectj.weaver.bcel.BcelWeaver.weave(BcelWeaver.java:1161)
[ERROR]       at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.weaveQueuedEntries(AjPipeliningCompilerAdapter.java:510)
[ERROR]       at org.aspectj.ajdt.internal.compiler.AjPipeliningCompilerAdapter.afterCompiling(AjPipeliningCompilerAdapter.java:374)
[ERROR]       at org.aspectj.ajdt.internal.compiler.CompilerAdapter.ajc$afterReturning$org_aspectj_ajdt_internal_compiler_CompilerAdapter$2$f9cc9ca0(CompilerAdapter.aj:79)
[ERROR]       at org.aspectj.org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:428)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performCompilation(AjBuildManager.java:1101)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.performBuild(AjBuildManager.java:275)
[ERROR]       at org.aspectj.ajdt.internal.core.builder.AjBuildManager.batchBuild(AjBuildManager.java:188)
[ERROR]       at org.aspectj.ajdt.ajc.AjdtCommand.doCommand(AjdtCommand.java:103)
[ERROR]       at org.aspectj.ajdt.ajc.AjdtCommand.runCommand(AjdtCommand.java:47)
[ERROR]       at org.aspectj.tools.ajc.Main.run(Main.java:374)
[ERROR]       at org.aspectj.tools.ajc.Main.runMain(Main.java:253)
[ERROR]       at org.codehaus.mojo.aspectj.AbstractAjcCompiler.execute(AbstractAjcCompiler.java:568)
[ERROR]       at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
[ERROR]       at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]       at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]       at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
[ERROR]       at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
[ERROR]       at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]       at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]       at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
[ERROR]       at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
[ERROR]       at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
[ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]       at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

My pom.xml:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>gov.nasa.drf</groupId>
    <artifactId>auth</artifactId>
    <version>2.0</version>

    <properties>
        <maven.compiler.source>17</maven.compiler.source>
        <maven.compiler.target>17</maven.compiler.target>
        <spring.version>6.1.0.RELEASE</spring.version>
        <aspectj.version>1.9.8</aspectj.version>
        <log4j2.version>2.17.1</log4j2.version>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.1.1</version>
    </parent>

    <repositories>
        <repository>
            <id>data-local</id>
            <name>data</name>
            <url>file://${project.basedir}/repo</url>
        </repository>
    </repositories>

    <build>
        <sourceDirectory>src/main/java</sourceDirectory>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-jar-plugin</artifactId>
                <configuration>
                    <archive>
                        <manifest>
                            <addClasspath>true</addClasspath>
                            <classpathPrefix>lib/</classpathPrefix>
                            <mainClass>gov.nasa.drf.auth.App</mainClass>
                        </manifest>
                    </archive>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <goals>
                            <goal>repackage</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.1.2</version>
                <dependencies>
                    <dependency>
                        <groupId>org.apache.maven.surefire</groupId>
                        <artifactId>surefire-junit47</artifactId>
                        <version>3.1.2</version>
                    </dependency>
                </dependencies>
                <configuration>
                    <parallel>both</parallel>
                    <threadCount>4</threadCount>
                    <includes>
                        <include>**/*Test</include>
                    </includes>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>aspectj-maven-plugin</artifactId>
                <version>1.14.0</version>
                <dependencies>
                    <dependency>
                        <groupId>org.aspectj</groupId>
                        <artifactId>aspectjrt</artifactId>
                        <version>${aspectj.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>org.aspectj</groupId>
                        <artifactId>aspectjtools</artifactId>
                        <version>${aspectj.version}</version>
                    </dependency>
                </dependencies>
                <executions>
                    <execution>
                        <id>default-compile</id>
                        <phase>process-classes</phase>
                        <goals>
                            <!-- use this goal to weave all your main classes -->
                            <goal>compile</goal>
                        </goals>
                        <configuration>
                            <weaveDirectories>
                                <weaveDirectory>${project.build.directory}/classes</weaveDirectory>
                            </weaveDirectories>
                        </configuration>
                    </execution>
                    <execution>
                        <id>default-testCompile</id>
                        <phase>process-test-classes</phase>
                        <goals>
                            <!-- use this goal to weave all your test classes -->
                            <goal>test-compile</goal>
                        </goals>
                        <configuration>
                            <weaveDirectories>
                                <weaveDirectory>${project.build.directory}/test-classes</weaveDirectory>
                            </weaveDirectories>
                        </configuration>
                    </execution>
                </executions>
                <configuration>
                    <outxml>true</outxml>
                    <verbose>true</verbose>
                    <showWeaveInfo>true</showWeaveInfo>
                    <aspectLibraries>
                        <aspectLibrary>
                            <groupId>org.springframework</groupId>
                            <artifactId>spring-aspects</artifactId>
                        </aspectLibrary>
                    </aspectLibraries>
                    <complianceLevel>17</complianceLevel>
                    <source>17</source>
                    <target>17</target>

                    <excludes>
                        <exclude>**/*.java</exclude>
                    </excludes>
                    <forceAjcCompile>true</forceAjcCompile>
                    <sources/>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.11.0</version>
                <configuration>
                    <annotationProcessorPaths>
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                            <version>1.18.28</version>
                        </path>
                    </annotationProcessorPaths>
                    <source>17</source>
                    <target>17</target>
                </configuration>
            </plugin>
        </plugins>
    </build>

    <dependencies>
        <dependency>
            <groupId>gov.nasa.drf</groupId>
            <artifactId>lib</artifactId>
            <version>1.0</version>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-jpa</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springdoc</groupId>
            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
            <version>2.0.2</version>
        </dependency>
        <dependency>
            <groupId>com.github.joschi.jackson</groupId>
            <artifactId>jackson-datatype-threetenbp</artifactId>
            <version>2.13.0-rc2</version>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.28</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.postgresql</groupId>
            <artifactId>postgresql</artifactId>
            <version>42.6.0</version>
        </dependency>
        <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjrt</artifactId>
            <version>${aspectj.version}</version>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjrt</artifactId>
            <version>${aspectj.version}</version>
        </dependency>
        <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjtools</artifactId>
            <version>${aspectj.version}</version>
        </dependency>
        <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjweaver</artifactId>
            <version>${aspectj.version}</version>
        </dependency>
        <dependency>
            <groupId>org.springframework</groupId>
            <artifactId>spring-aop</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework</groupId>
            <artifactId>spring-aspects</artifactId>
        </dependency>
    </dependencies>
</project>
```

I do not really understand the error. Is there a way to get my project to build with AspectJ and Maven? It runs in IntelliJ without issue.

--- a/tests/bugs1920/github_250/MyAspect.aj
+++ b/tests/bugs1920/github_250/MyAspect.aj
@@ -1,0 +1,40 @@
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+
+import java.util.Random;
+
+/**
+ * Reproduces <a href="https://github.com/eclipse-aspectj/aspectj/issues/250">Github bug 250</a>
+ */
+@Aspect
+public class MyAspect {
+  @Around("execution(* Application.*(..))")
+  public Object myAdvice1(ProceedingJoinPoint joinPoint) {
+    System.out.println(joinPoint);
+    a(1, "one");
+    return joinPoint.proceed();
+  }
+
+  @Around("execution(* Application.*(..))")
+  public Object myAdvice2(ProceedingJoinPoint joinPoint) throws Throwable {
+    System.out.println(joinPoint);
+    a(2);
+    return new Random().nextBoolean() ? joinPoint.proceed() : null;
+  }
+
+  private void a(int i) {
+    System.out.println(i);
+  }
+  private void a(int i, String s) {
+    System.out.println(i + " / " + s);
+  }
+
+  public static void main(String[] args) {
+    new Application().doSomething();
+  }
+}
+
+class Application {
+  public void doSomething() {}
+}

--- a/tests/java5/ataspectj/ajc-ant.xml
+++ b/tests/java5/ataspectj/ajc-ant.xml
@@ -107,7 +107,7 @@
     </target>
 
     <target name="ltwlog.LTWLog">
-        <javac target="1.5" destdir="${aj.sandbox}" classpathref="aj.path"
+        <javac target="1.8" destdir="${aj.sandbox}" classpathref="aj.path"
             srcdir="${basedir}"
             includes="ataspectj/ltwlog/*"
             debug="true">

--- a/tests/java5/ataspectj/ataspectj/AroundInlineMungerTestAspects.java
+++ b/tests/java5/ataspectj/ataspectj/AroundInlineMungerTestAspects.java
@@ -23,7 +23,9 @@ import org.aspectj.lang.ProceedingJoinPoint;
 public class AroundInlineMungerTestAspects {
 
     public static class OpenBase {
-        protected void superMethod() {}
+        protected String superMethod(String s) {
+            return s;
+        }
     }
 
     public static class OpenSubBase extends OpenBase {}
@@ -43,7 +45,7 @@ public class AroundInlineMungerTestAspects {
         public Object around1(ProceedingJoinPoint jp) throws Throwable {
             aroundCount++;
             priv(1, 2L, 3);
-            super.superMethod();
+            super.superMethod("x");
             new Open.Inner().priv();//fails to be wrapped so this advice will not be inlined but previous call were still prepared
             return jp.proceed();
         }
@@ -64,7 +66,7 @@ public class AroundInlineMungerTestAspects {
         @SuppressAjWarnings
         public Object around2(ProceedingJoinPoint jp) throws Throwable {
             aroundCount++;
-            super.superMethod();
+            super.superMethod("x");
             new Open.Inner().priv();//fails to be wrapped so next calls won't be prepared but previous was
             priv(1, 2L, 3);
             return jp.proceed();

--- a/tests/src/test/java/org/aspectj/systemtest/ajc1920/Bugs1920Tests.java
+++ b/tests/src/test/java/org/aspectj/systemtest/ajc1920/Bugs1920Tests.java
@@ -26,6 +26,15 @@ public class Bugs1920Tests extends XMLBasedAjcTestCase {
     runTest("add correct annotations to multiple ITD methods with the same name and same number of arguments");
   }
 
+  /**
+   * Make sure to create one {@code ajc$inlineAccessMethod} for identically named (overloaded) private aspect methods.
+   * <p>
+   * See <a href="https://github.com/eclipse-aspectj/aspectj/issues/250">GitHub issue 250</a>.
+   */
+  public void test_GitHub_250() {
+    runTest("correctly handle overloaded private methods in aspects");
+  }
+
   public static Test suite() {
     return XMLBasedAjcTestCase.loadSuite(Bugs1920Tests.class);
   }

--- a/tests/src/test/resources/org/aspectj/systemtest/ajc1920/ajc1920.xml
+++ b/tests/src/test/resources/org/aspectj/systemtest/ajc1920/ajc1920.xml
@@ -190,4 +190,17 @@
 		</run>
 	</ajc-test>
 
+	<!-- https://github.com/eclipse-aspectj/aspectj/issues/250 -->
+	<ajc-test dir="bugs1920/github_250" vm="8" title="correctly handle overloaded private methods in aspects">
+		<compile files="MyAspect.aj" options="-8"/>
+		<run class="MyAspect">
+			<stdout>
+				<line text="execution(void Application.doSomething())"/>
+				<line text="1 / one"/>
+				<line text="execution(void Application.doSomething())"/>
+				<line text="2"/>
+			</stdout>
+		</run>
+	</ajc-test>
+
 </suite>

--- a/weaver/src/main/java/org/aspectj/weaver/bcel/BcelAccessForInlineMunger.java
+++ b/weaver/src/main/java/org/aspectj/weaver/bcel/BcelAccessForInlineMunger.java
@@ -101,7 +101,11 @@ public class BcelAccessForInlineMunger extends BcelTypeMunger {
 	 */
 	@Override
 	public ResolvedMember getMatchingSyntheticMember(Member member) {
-		ResolvedMember rm = inlineAccessors.get(member.getName());// + member.getSignature());
+    String name = member.getName();
+    String signature = name.startsWith("ajc$superDispatch$")
+      ? member.getSignature()
+      : member.getSignature().replaceFirst("\\([^;]+;", "(");
+    ResolvedMember rm = inlineAccessors.get(name + signature);
 //		System.err.println("lookup for " + member.getName() + ":" + member.getSignature() + " = "
 //				+ (rm == null ? "" : rm.getName()));
 		return rm;
@@ -229,7 +233,7 @@ public class BcelAccessForInlineMunger extends BcelTypeMunger {
 	private ResolvedMember createOrGetInlineAccessorForMethod(ResolvedMember resolvedMember) {
 		String accessorName = NameMangler.inlineAccessMethodForMethod(resolvedMember.getName(), resolvedMember.getDeclaringType(),
 				aspectType);
-		String key = accessorName;// new StringBuilder(accessorName).append(resolvedMember.getSignature()).toString();
+		String key = accessorName + resolvedMember.getSignature();
 		ResolvedMember inlineAccessor = inlineAccessors.get(key);
 //		System.err.println(key + " accessor=" + inlineAccessor);
 		if (inlineAccessor == null) {
@@ -272,7 +276,7 @@ public class BcelAccessForInlineMunger extends BcelTypeMunger {
 	private ResolvedMember createOrGetInlineAccessorForSuperDispatch(ResolvedMember resolvedMember) {
 		String accessor = NameMangler.superDispatchMethod(aspectType, resolvedMember.getName());
 
-		String key = accessor;
+		String key = accessor + resolvedMember.getSignature();
 		ResolvedMember inlineAccessor = inlineAccessors.get(key);
 
 		if (inlineAccessor == null) {
@@ -316,7 +320,7 @@ public class BcelAccessForInlineMunger extends BcelTypeMunger {
 	private ResolvedMember createOrGetInlineAccessorForFieldGet(ResolvedMember resolvedMember) {
 		String accessor = NameMangler.inlineAccessMethodForFieldGet(resolvedMember.getName(), resolvedMember.getDeclaringType(),
 				aspectType);
-		String key = accessor;
+		String key = accessor + "()" + resolvedMember.getSignature();
 		ResolvedMember inlineAccessor = inlineAccessors.get(key);
 
 		if (inlineAccessor == null) {
@@ -357,7 +361,7 @@ public class BcelAccessForInlineMunger extends BcelTypeMunger {
 	private ResolvedMember createOrGetInlineAccessorForFieldSet(ResolvedMember resolvedMember) {
 		String accessor = NameMangler.inlineAccessMethodForFieldSet(resolvedMember.getName(), resolvedMember.getDeclaringType(),
 				aspectType);
-		String key = accessor;
+		String key = accessor + "(" + resolvedMember.getSignature() + ")V";
 		ResolvedMember inlineAccessor = inlineAccessors.get(key);
 
 		if (inlineAccessor == null) {

--- a/weaver/src/main/java/org/aspectj/weaver/bcel/LazyClassGen.java
+++ b/weaver/src/main/java/org/aspectj/weaver/bcel/LazyClassGen.java
@@ -753,14 +753,20 @@ public final class LazyClassGen {
 		// are required (unless turning off the verifier)
 		if ((myGen.getMajor() == Constants.MAJOR_1_6 && world.shouldGenerateStackMaps()) || myGen.getMajor() > Constants.MAJOR_1_6) {
 			if (!AsmDetector.isAsmAround) {
-				// Fix https://github.com/eclipse-aspectj/aspectj/issues/251, using "replace('Ä', 'Ö')" to avoid non-relocated
-				// class names to be embedded into the error message during compile time, making it end up in the class constant
-				// pool unwantedly, as this clashes with post-compile ASM package relocation.
-				final String errorMessage = "Unable to find ASM classes (" +
-					AsmDetector.CLASS_READER.replace('Ä', 'Ö') + ", " + AsmDetector.CLASS_VISITOR.replace('Ä', 'Ö') + ") " +
-					"for stackmap generation. Stackmap generation for woven code is required to avoid verify errors " +
-					"on a Java 1.7 or higher runtime.";
-				throw new BCException(errorMessage, AsmDetector.reasonAsmIsMissing);
+				if (
+					AsmDetector.rootCause instanceof ClassNotFoundException ||
+						AsmDetector.rootCause instanceof NoClassDefFoundError
+				) {
+					// Fix https://github.com/eclipse-aspectj/aspectj/issues/251, using "replace('Ä', 'Ö')" to avoid
+					// non-relocated class names to be embedded into the error message during compile time, making it end up in
+					// the class constant pool unwantedly, as this clashes with post-compile ASM package relocation.
+					String errorMessage = "Unable to find ASM classes (" +
+						AsmDetector.CLASS_READER.replace('Ä', 'Ö') + ", " + AsmDetector.CLASS_VISITOR.replace('Ä', 'Ö') + ") " +
+						"for stackmap generation. Stackmap generation for woven code is required to avoid verify errors " +
+						"on a Java 1.7 or higher runtime.";
+					throw new BCException(errorMessage, AsmDetector.rootCause);
+				}
+				throw new BCException("Error processing class file", AsmDetector.rootCause);
 			}
 			wovenClassFileData = StackMapAdder.addStackMaps(world, myGen.getClassName(), wovenClassFileData);
 		}

--- a/weaver/src/main/java/org/aspectj/weaver/bcel/asm/AsmDetector.java
+++ b/weaver/src/main/java/org/aspectj/weaver/bcel/asm/AsmDetector.java
@@ -20,7 +20,7 @@ public class AsmDetector {
 	public static final String CLASS_READER = "org.objectweb.asm.ClassReader";
 	public static final String CLASS_VISITOR = "org.objectweb.asm.ClassVisitor";
 	public static boolean isAsmAround;
-	public static Throwable reasonAsmIsMissing;
+	public static Throwable rootCause;
 
 	static {
 		try {
@@ -30,7 +30,7 @@ public class AsmDetector {
 			isAsmAround = true;
 		} catch (Exception e) {
 			isAsmAround = false;
-			reasonAsmIsMissing = e;
+			rootCause = e;
 		}
 		//System.out.println(isAsmAround ? "ASM detected" : "No ASM found");
 	}

--- a/weaver/src/main/java/org/aspectj/weaver/bcel/asm/StackMapAdder.java
+++ b/weaver/src/main/java/org/aspectj/weaver/bcel/asm/StackMapAdder.java
@@ -49,7 +49,7 @@ public class StackMapAdder {
 			System.err.println("AspectJ Internal Error: unable to add stackmap attributes to class '"+classname+"'. " + t.getMessage());
 			t.printStackTrace();
 			AsmDetector.isAsmAround = false;
-			AsmDetector.reasonAsmIsMissing = t;
+			AsmDetector.rootCause = t;
 			return data;
 		}
 	}

